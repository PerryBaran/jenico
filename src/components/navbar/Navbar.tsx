import { Link } from "react-router-dom";
import style from "./navbar.module.css";

function Navbar() {
  return (
    <nav className={style.container}>
      <Link to="/" className={style.link}>
        Jenico
      </Link>
      <div />
      <Link to="/" className={style.link}>
        Home
      </Link>
      <div />
      <Link to="/music" className={style.link}>
        Music
      </Link>
      <div />
      <Link to="/contact" className={style.link}>
        Contact
      </Link>
    </nav>
  );
}

export default Navbar;
