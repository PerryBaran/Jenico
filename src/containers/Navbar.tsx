import React from 'react';
import { Link } from 'react-router-dom';


function Navbar(props:any) {
    return (
        <div className='navbar'>
            <h1>Jenico</h1>
            <Link to="/" className='navLink'>
                <button>Home</button>
            </Link>
            <Link to="/music" className='navLink'>
                <button>Music</button>
            </Link>
            <Link to="/contact" className='navLink'>
                <button>Contact</button>
            </Link>
        </div>
    );
}

export default Navbar;