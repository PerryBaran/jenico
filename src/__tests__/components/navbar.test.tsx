import { render, screen, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import Navbar from "../../components/navbar/Navbar";
import { BrowserRouter as Router } from "react-router-dom";

const RenderWithRouter = () => {
  return (
    <Router>
      <Navbar />
    </Router>
  );
};

describe("Navbar", () => {
  afterEach(cleanup);

  test("snapshot", () => {
    const { asFragment } = render(<RenderWithRouter />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("props", () => {
    render(<RenderWithRouter />);

    expect(screen.getByText(/jenico/i)).toHaveAttribute("href", "/");
    expect(screen.getByText(/home/i)).toHaveAttribute("href", "/");
    expect(screen.getByText(/music/i)).toHaveAttribute("href", "/music");
    expect(screen.getByText(/contact/i)).toHaveAttribute("href", "/contact");
  });
});
