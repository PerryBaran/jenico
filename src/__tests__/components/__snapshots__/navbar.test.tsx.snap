// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar snapshot 1`] = `
<DocumentFragment>
  <nav
    class="container"
  >
    <a
      class="link"
      href="/"
    >
      Jenico
    </a>
    <div />
    <a
      class="link"
      href="/"
    >
      Home
    </a>
    <div />
    <a
      class="link"
      href="/music"
    >
      Music
    </a>
    <div />
    <a
      class="link"
      href="/contact"
    >
      Contact
    </a>
  </nav>
</DocumentFragment>
`;
